### PR TITLE
Add handlers

### PR DESCRIPTION
**Pull request by Widcket**
### Description

This PR adds implementations for the apps, APIs, and roles handlers. Conflict resolution was not implemented.

https://user-images.githubusercontent.com/5055789/157835567-64759281-21bd-4556-85c3-ec93c5ad7918.mov

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

File: internal/cli/import_yaml.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+			errMsg := fmt.Sprintf("found duplicate name in resourceServer: %s", rs.Name)
+			result = multierror.Append(result, errors.New(errMsg))
+		}
+		resourceServerNames[rs.Name] = true
+	}
+
+	roleNames := make(map[string]bool)
+	for _, r := range t.Roles {
+		if roleNames[r.Name] {
+			errMsg := fmt.Sprintf("found duplicate name in resourceServer: %s", r.Name)
```
Comment (christonog) [inline internal/cli/import_yaml.go:214]: Probably missed by me earlier, but should change to `found duplicate name in role`
```suggestion
			errMsg := fmt.Sprintf("found duplicate name in role: %s", r.Name)
```

Comment (Widcket) [reply internal/cli/import_yaml.go]: Thank you!

File: internal/cli/import_yaml.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+		}
+
+		for j, item := range client.AllowedOrigins {
+			if strings.ContainsAny(item, key) {
+				str := strings.ReplaceAll(item, fmt.Sprintf("##%s##", key), replacement.(string))
+				t.Clients[i].AllowedOrigins[j] = str
+			}
+		}
+
+		for j, item := range client.GrantTypes {
```
Comment (christonog) [inline internal/cli/import_yaml.go:130]: I see we added more fields for clients and apis. Do we know what fields are configurable? Or can we allow replacements for all fields?

File: internal/cli/import_roles.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+// 4. Perform patch diff
+// 5. Execute creates, updates and deletes
+
+// 1. Fetch all existing items
+
+func (cli *cli) getAllRoles(context context.Context) ([]*Role, error) {
+	list, err := getWithPagination(
+		context,
+		0, // Get *all* roles
+		func(opts ...management.RequestOption) (result []interface{}, hasNext bool, apiErr error) {
```
Comment (christonog) [inline internal/cli/import_roles.go:28]: looks like `opts` is available with the context that is passed in?

File: internal/cli/import_yaml.go (hunk 2 above)
Comment (Widcket) [reply internal/cli/import_yaml.go:130]: Thanks for pointing this out. I compared with the Management API docs, and ended adding `omitempty` for `app_type` (apps) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/01280ba29f28d6a86c0af685ee724dbbc0df9d05)), `name` (APIs) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/5387d55e3faf9d1b5faaa1a183b812a265922b09)), and `description` (roles) ([commit](https://github.com/auth0/auth0-cli/pull/404/commits/f50b7fd53224046e40fc202f5d82a8a771dea1f8)).

**Apps:** only `name` is required.
Screen Shot 2022-03-11 at 12 31 06

**APIs:** only `identifier` is required.
Screen Shot 2022-03-11 at 12 31 50

**Roles:** only `name` is required.
Screen Shot 2022-03-11 at 12 32 15

File: internal/cli/import_roles.go (hunk 3 above)
Comment (christonog) [reply internal/cli/import_roles.go:28]: I guess I'm curious where opts is passed in to `cli.getAllAPIs(ctx)` since only a ctx is passed in?

**Review by christonog — APPROVED** [review]
@Widcket Do you think you could share your `tenant.yaml` and non-secret `config.json` so I can give it a go on my end as well? Curious to see how it looks in the management dashboard.

**Comment by Widcket** [comment]
> @Widcket Do you think you could share your `tenant.yaml` and non-secret `config.json` so I can give it a go on my end as well? Curious to see how it looks in the management dashboard.

Sure, I'll send via PM.

File: internal/cli/import_roles.go (hunk 3 above)
Comment (Widcket) [reply internal/cli/import_roles.go:28]: Oh, the func takes opts but none are passed of course. This is copypasta from https://github.com/auth0/auth0-cli/blob/main/internal/cli/organizations.go#L688.

**Pull request merged**